Source/Common/MemoryCommon.cpp

```diff
@@ -490,11 +490,20 @@ std::string formatMemoryToString(const char* memory, const MemType type, const s
       std::memcpy(memoryCopy, &word, sizeof(u32));
     }
 
-    float aFloat = 0.0f;
-    std::memcpy(&aFloat, memoryCopy, sizeof(float));
-    // With 9 digits of precision, it is possible to convert a float back and forth to its binary
-    // representation without any loss
-    ss << std::setprecision(9) << aFloat;
+    if (base == Common::MemBase::base_hexadecimal)
+    {
+      u32 unsignedWord = 0;
+      std::memcpy(&unsignedWord, memoryCopy, sizeof(u32));
+      ss << unsignedWord;
+    }
```

Comment:
While we are here, I think we can also support the other representations (octal and binary). Suggestion:

```cpp
    if (base != Common::MemBase::base_decimal)
    {
      u32 unsignedWord = 0;
      std::memcpy(&unsignedWord, memoryCopy, sizeof(u32));
      if (base == Common::MemBase::base_binary)
      {
        ss << std::bitset<sizeof(u32) * 8>(unsignedWord).to_string();
      }
      else
      {
        ss << unsignedWord;
      }
    }
    else
    {
      float aFloat = 0.0f;
      ...
    }
```

I think binary in particular can be useful.

(Similar change for the `double` type below.)

Reply:
Sure thing. I also noticed that I did not account for when the user types in a new float value in hex, so I will need to make appropriate changes for that too.